File: roboplan_examples/examples/test_scene.cpp
#include <filesystem>
#include <vector>

#include <examples/example_resources.hpp>
#include <roboplan/core/scene.hpp>

int main(int /*argc*/, char* /*argv*/[]) {

  const auto share_prefix = roboplan_examples::get_package_share_dir();

  // Set up the scene
  const auto urdf_path = share_prefix / "ur_robot_model" / "ur5_gripper.urdf";
  const auto srdf_path = share_prefix / "ur_robot_model" / "ur5_gripper.srdf";
  const std::vector<std::filesystem::path> package_paths = {share_prefix};

  auto scene =
      roboplan::Scene("test_scene", urdf_path, srdf_path, package_paths);
  scene.print();

  return 0;
}

File: bindings/examples/test_scene.py
from pathlib import Path

import numpy as np

from utils import get_example_resources_directory, get_package_path

import roboplan

if __name__ == "__main__":

    jc = roboplan.JointConfiguration(
        ["joint_1", "joint_2", "joint_3"],
        np.array([0.1, 0.2, 0.3]),
    )
    print(f"Config names: {jc.joint_names}")
    print(f"Config positions: {jc.positions}")
    print("")

    roboplan_examples_dir = Path(get_example_resources_directory())
    urdf_path = roboplan_examples_dir / "ur5_gripper.urdf"
    srdf_path = roboplan_examples_dir / "ur5_gripper.srdf"
    package_paths = [get_package_path()]

    scene = roboplan.Scene("test_scene", urdf_path, srdf_path, package_paths)
    scene.print()
